LICENSE-MIT

```diff
@@ -1,3 +1,6 @@
+This software includes components from third parties. For a full list of these components and their licenses,
+see the THIRD PARTY NOTICES file.
+
 MIT License
 
 Copyright (c) 2019 Matter Labs
```

Comment:
You should add plonky2 copyright here instead of adding "notice" to the header.

Reply:
@paulmillr we discussed the right pattern to follow here, and ultimately this seems to be one that can keep a relatively clean top-level license file, and add all of the appropriate attribution for third party development. 

This is a pretty established approach, as best we understand it. 

e.g. https://github.com/microsoft/vscode/blob/main/ThirdPartyNotices.txt

Reply:
The thirdpartynotices you've linked is a list of all dependencies. You don't depend on plonky2, it's not your third party lib.

Reply:
So you have a derived work, not a third party dependency.

Reply:
https://opensource.stackexchange.com/questions/4506/what-license-do-i-apply-to-my-software-if-parts-of-its-code-is-derived-from-mit

Reply:
Thanks, we've confirmed with legal counsel that our approach is valid.

Reply:
awesome